### PR TITLE
ci.ivytest-documentTest

### PR DESCRIPTION
**Pull request by nhthinh-axonivy**
- Create test cases for document detectors and SVGSecurityScanner
- Update SVGSecurityScanner to handle edge case

### CODE REVIEW

File: AxonIvyPortal/portal-components/src/com/axonivy/portal/components/document/SVGSecurityScanner.java [hunk 1]
```diff
@@ -93,7 +93,8 @@ public static boolean isSafe(String svgContent) {
       }
 
       for (Element style : doc.select("style")) {
-        String cssLower = style.data().toLowerCase(Locale.ROOT);
+        String cssData = style.data();
+        String cssLower = (cssData.isEmpty() ? style.wholeOwnText() : cssData).toLowerCase(Locale.ROOT);
```
Comment (ivy-rew) [inline AxonIvyPortal/portal-components/src/com/axonivy/portal/components/document/SVGSecurityScanner.java:97]: good catch; testing helps 😉

File: AxonIvyPortal/portal-test/src_test/com/axonivy/portal/document/TestDocumentDetectorFactory.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"pdf"})
+  void returnsPDFDetector(String extension) {
+    assertThat(factory.getDocumentDetector(extension)).isInstanceOf(PDFDocumentDetector.class);
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"xls", "xlsx", "xlsm", "xlsb", "xlt", "xltm"})
```
Comment (ivy-rew) [inline AxonIvyPortal/portal-test/src_test/com/axonivy/portal/document/TestDocumentDetectorFactory.java:36]: great example of ParameterizedTest ; let's share it with the team @axonivy-market/team-wawa

File: AxonIvyPortal/portal-test/src_test/com/axonivy/portal/document/TestWordDocumentDetector.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  @Test
+  void corruptedStream_isSafe() {
+    // Unreadable content: IOException path → containsMacrosOrActiveX = false → safe
+    byte[] garbage = "not-a-word-file".getBytes();
+    assertThat(detector.isSafe(new ByteArrayInputStream(garbage))).isTrue();
+  }
+
+  //TODO:Testing "has macros → not safe" requires a real OLE2 (.doc/.docm) file with embedded VBA or ActiveX controls.
```
Comment (ivy-rew) [inline AxonIvyPortal/portal-test/src_test/com/axonivy/portal/document/TestWordDocumentDetector.java]: I'd actually check-in a real small file which is in this state -> then load it. E.g. `TestWordDocumentDetectorr.class.getResourceAsStream("bad.docm")`

**Review by ivy-rew — DISMISSED** [review]
Very nice

**Review by ivy-rew — APPROVED** [review]
perfect 🤠

**Comment by nhthinh-axonivy** [comment]
> perfect 🤠

Thanks for the approval! Otherwise, this PR might have been buried in the wasteland for a while. 😄
